tests/all_bad_test.py

```diff
@@ -13,7 +13,9 @@ class TestAllBad(object):
         ")",
         # "a / b /",
         "\\frac{d}{dx}",
-        "(\\frac{d}{dx})"
+        "(\\frac{d}{dx})",
+        "\\frac{d}{dx}^2 x",
```

Comment:
just out of curiosity, where did the new test cases come from? things you thought of, from other forks, existing problem instances?

Reply:
@lily514 honestly it was trial and error trying to figure out how to reach each of the parser/code branches